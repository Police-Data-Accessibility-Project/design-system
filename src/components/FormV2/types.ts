// TODO: remove the V2 from all of these types when Form is removed and FormV2 -> Form

import useVuelidate from '@vuelidate/core';
import { makeRules } from './util';
import { Ref } from 'vue';

import * as validators from '@vuelidate/validators';

export interface PdapFormValidatorV2<T> {
	message?: string;
	value?: T;
	predicate?: (value: T) => boolean;
}

export type PdapFormValidatorsV2 = Record<
	keyof typeof validators,
	PdapFormValidatorV2<string | number | boolean>
>;

export type ValidationSchemaV2 = {
	name: string;
	validators: Partial<PdapFormValidatorsV2>;
}[];
/**
 * PDAP Form props interface.
 */
export interface PdapFormPropsV2 {
	defaultValues?: Record<PropertyKey, unknown>;
	error?: string | undefined | null;
	// Adds id and name in order to make required
	id: string;
	name: string;
	schema: ValidationSchemaV2;
}

export interface PdapFormProvideV2 {
	values: Ref<Record<PropertyKey, unknown>>;
	setValues: (values: Record<PropertyKey, unknown>) => void;
	rules: ReturnType<typeof makeRules>;
	v$: ReturnType<typeof useVuelidate>;
}
