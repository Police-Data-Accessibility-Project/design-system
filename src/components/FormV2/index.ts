export { default as FormV2 } from './PdapFormV2.vue';
