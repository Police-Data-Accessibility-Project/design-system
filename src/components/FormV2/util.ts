import { createRule } from '../../utils/vuelidate';
import { PdapFormProvideV2, ValidationSchemaV2 } from './types';
import { InjectionKey } from 'vue';

export function makeRules(
	schema: ValidationSchemaV2
): Record<string, ReturnType<typeof createRule>> {
	return schema.reduce((acc, { name, validators }) => {
		const toAdd = Object.entries(validators ?? {}).reduce((acc, [key, val]) => {
			return {
				...acc,
				...createRule(key, val),
			};
		}, {});

		return {
			...acc,
			[name]: {
				...toAdd,
			},
		};
	}, {});
}

export const provideKey = Symbol() as InjectionKey<PdapFormProvideV2>;
