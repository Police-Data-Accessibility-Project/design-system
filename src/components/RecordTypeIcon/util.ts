/**
 * Creates map with type inference
 *
 * UNUSED FOR NOW, but we may need it later
 */
// export function ReadonlyMapWithStringKeys<K extends string, V>(
// 	iterable: Iterable<[K, V]>
// ): ReadonlyMap<K, V> {
// 	return new Map(iterable);
// }

// type MapKeys = typeof <<map>> extends ReadonlyMap<infer K, any>
// 	? K
// 	: never;
