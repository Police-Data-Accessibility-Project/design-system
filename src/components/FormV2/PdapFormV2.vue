<template>
	<form
		:id="id"
		:name="name"
		class="pdap-form"
		@submit.prevent="submit"
		@input.prevent="change"
		@change.prevent="change"
	>
		<slot v-if="$slots.error" name="error" />
		<div
			v-else-if="typeof errorMessage === 'string'"
			class="pdap-form-error-message"
		>
			{{ errorMessage }}
		</div>

		<slot />
	</form>
</template>
<script setup lang="ts">
// Globals
import { provide, ref, watchEffect } from 'vue';
import { useVuelidate } from '@vuelidate/core';

// Types
import { PdapFormPropsV2 } from './types';
import { provideKey, makeRules } from './util';

// Props
const props = defineProps<PdapFormPropsV2>();

// Emits
const emit = defineEmits(['submit', 'change', 'error']);

// Constants
const errorMessage = ref(props.error);
const values = ref(props.defaultValues ?? {});
const rules = props.schema ? makeRules(props.schema) : {};
const v$ = useVuelidate(rules, values, { $autoDirty: false, $lazy: true });

// Provide
provide(provideKey, {
	setValues,
	values,
	rules,
	v$,
});

// Expose
defineExpose({
	setValues,
	v$,
});

// Helpers
function setValues(val: Partial<typeof values.value>) {
	values.value = { ...values.value, ...val };
}
function resetForm() {
	v$.value.$reset();
	values.value = Object.entries(values.value).reduce((acc, [key]) => {
		return {
			...acc,
			[key]:
				typeof values.value[key] === 'string'
					? ''
					: Boolean(props.defaultValues?.[key]),
		};
	}, {});
}
function change(event: Event) {
	emit('change', values.value, event);
}

async function submit(e: Event) {
	// Check form submission
	const isValidSubmission = await v$.value.$validate();
	if (!isValidSubmission) emit('error', v$);
	if (isValidSubmission) {
		// Emit submit event (spread to new object to create new object, this allows us to reset `values` without messing with the data returned)
		emit('submit', { ...values.value }, e);

		resetForm();
	}
}

// Effects
// Effect - Updates form error state based on input error state and/or props
watchEffect(() => {
	if (props.error) errorMessage.value = props.error;
	else if (errorMessage.value && v$.value.$errors.length === 0)
		errorMessage.value = null;
	else if (!errorMessage.value && v$.value.$errors.length > 0)
		errorMessage.value = 'Please update this form to correct the errors';
});
</script>

<style>
@tailwind components;

@layer components {
	.pdap-form-error-message {
		@apply items-center justify-start basis-full flex-shrink flex bg-red-300 text-red-800 mb-3 p-2 text-sm;
	}
}
</style>
