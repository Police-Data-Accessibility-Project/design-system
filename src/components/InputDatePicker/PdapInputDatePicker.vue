<template>
	<div class="pdap-input" :class="{ 'pdap-input-error': error }">
		<label v-if="$slots.label" :id="`${name}-${id}-label`" :for="id">
			<slot name="label" />
		</label>
		<label v-else-if="label" :id="`${name}-${id}-label`" :for="id">
			{{ label }}
		</label>
		<div v-if="$slots.error && error" class="pdap-input-error-message">
			<slot name="error" />
		</div>
		<div v-else-if="error" class="pdap-input-error-message">{{ error }}</div>

		<VueDatePicker
			v-bind="{ ...$attrs, ...$props }"
			v-model="date"
			:state="Boolean(error)"
			:dark="darkModePreference"
		/>
	</div>
</template>

<script setup lang="ts">
import { computed, inject, onMounted, onUnmounted, ref, watch } from 'vue';
import VueDatePicker from '@vuepic/vue-datepicker';
import '@vuepic/vue-datepicker/dist/main.css';
import { provideKey } from '../FormV2/util';
import { PdapFormProvideV2 } from '../FormV2/types';
import { PdapDatePickerProps } from './types';
import _isEqual from 'lodash/isEqual';

const { setValues, values, v$ } = inject<PdapFormProvideV2>(provideKey)!;
const { label, name, id } = defineProps<PdapDatePickerProps>();

const date = ref();
const error = computed(() => v$.value[name]?.$errors?.[0]?.$message);

// TODO: when decision made on whether to do dark them with inputs, pass as `:dark` prop to `VueDatePicker` (also add listener for updates )
const darkModeQuery = ref(window.matchMedia('(prefers-color-scheme: dark)'));
const darkModePreference = ref(darkModeQuery.value.matches);

function updateColorMode(e: MediaQueryListEvent) {
	darkModePreference.value = e.matches;
}
onMounted(() => {
	darkModeQuery.value.addEventListener('change', updateColorMode);
});
onUnmounted(() => {
	darkModeQuery.value.removeEventListener('change', updateColorMode);
});

watch(
	() => date.value,
	(newDate) => {
		// Sync values with underlying input, for form
		if (newDate) setValues({ [name]: newDate });
	}
);

watch(
	// Welcome to the land of edge-cases.
	() => values.value,
	// In the (unlikely, unrecommended, but sometimes unfortunately necessary) event of form values changing upstream from a parent component:
	(formValuesUpdated) => {
		const val = formValuesUpdated[name];
		// Case 0: Values are equivalent, or the change was made here, do nothing.

		/*
		 * Case 1: Value does not exist in form values object, meaning either:
		 ** a. it has not been set by the upstream change, or
		 ** b. has been changed to an empty string by `Form` after submit event
		 ** In either case, clear state or keep it clear
		 */
		if (!val) date.value = undefined;
		// Case 2 (rare): value has been programmatically updated upstream of `Form`
		else if (
			!_isEqual(val, date.value) &&
			(val instanceof Date ||
				(Array.isArray(val) && val.every((date) => date instanceof Date)))
		) {
			// Set the date to the value of the form value
			date.value = formValuesUpdated[name];
		}
	}
);
</script>

<style>
@tailwind base;

.dp__theme_light,
.dp__theme_dark {
	/* FULL LIST OF CUSTOMIZING OPTIONS from https://vue3datepicker.com/customization/theming/ */
	/* --dp-background-color: #212121;
    --dp-text-color: #fff;
    --dp-hover-color: #484848;
    --dp-hover-text-color: #fff;
    --dp-hover-icon-color: #959595;
    --dp-primary-color: #005cb2;
    --dp-primary-disabled-color: #61a8ea;
    --dp-primary-text-color: #fff;
    --dp-secondary-color: #a9a9a9;
    --dp-border-color: #2d2d2d;
    --dp-menu-border-color: #2d2d2d;
    --dp-border-color-hover: #aaaeb7;
    --dp-border-color-focus: #aaaeb7;
    --dp-disabled-color: #737373;
    --dp-disabled-color-text: #d0d0d0;
    --dp-scroll-bar-background: #212121;
    --dp-scroll-bar-color: #484848;
    --dp-success-color: #00701a;
    --dp-success-color-disabled: #428f59;
    --dp-icon-color: #959595;
    --dp-danger-color: #e53935;
    --dp-marker-color: #e53935;
    --dp-tooltip-color: #3e3e3e;
    --dp-highlight-color: rgb(0 92 178 / 20%);
    --dp-range-between-dates-background-color: var(--dp-hover-color, #484848);
    --dp-range-between-dates-text-color: var(--dp-hover-text-color, #fff);
    --dp-range-between-border-color: var(--dp-hover-color, #fff); */
}

.dp__theme_light {
	--dp-primary-color: rgb(var(--color-brand-gold-600));
	--dp-primary-text-color: rgb(var(--color-neutral-50));
}

.dp__theme_dark {
	--dp-primary-color: rgb(var(--color-brand-gold-700));
	--dp-primary-text-color: rgb(var(--color-neutral-950));
}

.pdap-input .dp__input {
	@apply text-lg pl-9;
}

div[role='gridcell'] {
	@apply outline-none border-2 border-solid border-transparent;
}

div[role='gridcell']:hover,
div[role='gridcell']:focus,
div[role='gridcell']:focus-visible {
	@apply border-2 border-brand-gold-300 border-solid outline-none;
}
</style>
