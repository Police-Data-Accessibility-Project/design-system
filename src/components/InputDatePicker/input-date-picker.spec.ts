import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
import { mount, VueWrapper } from '@vue/test-utils';
import PdapInputDatePicker from './PdapInputDatePicker.vue';
import { provideKey } from '../FormV2/util';
import VueDatePicker from '@vuepic/vue-datepicker';
import { ref } from 'vue';

describe('PdapInputDatePicker', () => {
	let wrapper: VueWrapper;
	const mockSetValues = vi.fn();
	const mockValues = ref({});
	const mockV$ = ref({
		testName: {
			$error: false,
			$errors: [],
		},
	});

	const defaultProps = {
		name: 'testName',
		id: 'test-id',
		label: 'Test Label',
	};

	const removeEventListener = vi.fn();
	// Mock window.matchMedia
	const mockMatchMedia = vi.fn().mockImplementation((query) => ({
		matches: false,
		media: query,
		onchange: null,
		addEventListener: vi.fn(),
		removeEventListener,
		dispatchEvent: vi.fn(),
	}));

	beforeEach(() => {
		global.MediaQueryListEvent = vi
			.fn()
			.mockImplementation((type, eventInitDict) => ({
				type,
				matches: eventInitDict.matches,
				media: '',
				target: {
					matches: eventInitDict.matches,
				},
			}));

		window.matchMedia = mockMatchMedia;
		wrapper = mount(PdapInputDatePicker, {
			props: defaultProps,
			global: {
				provide: {
					[provideKey as symbol]: {
						setValues: mockSetValues,
						values: mockValues,
						v$: mockV$,
					},
				},
				stubs: {
					VueDatePicker: true,
				},
			},
		});
	});

	afterEach(() => {
		vi.clearAllMocks();
		vi.unstubAllGlobals();
	});

	describe('Rendering', () => {
		it('renders correctly with default props', () => {
			expect(wrapper.exists()).toBe(true);
			expect(wrapper.find('label').text()).toBe('Test Label');
			expect(wrapper.findComponent(VueDatePicker).exists()).toBe(true);
		});

		it('renders label prop when no slot is provided', () => {
			wrapper = mount(PdapInputDatePicker, {
				props: {
					...defaultProps,
					label: 'Label from prop',
				},
				global: {
					provide: {
						[provideKey as symbol]: {
							setValues: mockSetValues,
							values: mockValues,
							v$: mockV$,
						},
					},
					stubs: {
						VueDatePicker: true,
					},
				},
			});

			const label = wrapper.find(`label[for="${defaultProps.id}"]`);
			expect(label.exists()).toBe(true);
			expect(label.text()).toBe('Label from prop');
			expect(label.attributes('id')).toBe(
				`${defaultProps.name}-${defaultProps.id}-label`
			);
		});

		it('renders label slot when provided', () => {
			wrapper = mount(PdapInputDatePicker, {
				props: {
					...defaultProps,
					label: 'Label from prop', // This should be ignored when slot is present
				},
				slots: {
					label: '<span class="custom-label">Custom Label Content</span>',
				},
				global: {
					provide: {
						[provideKey as symbol]: {
							setValues: mockSetValues,
							values: mockValues,
							v$: mockV$,
						},
					},
					stubs: {
						VueDatePicker: true,
					},
				},
			});

			const label = wrapper.find(`label[for="${defaultProps.id}"]`);
			expect(label.exists()).toBe(true);
			expect(label.find('.custom-label').exists()).toBe(true);
			expect(label.text()).toBe('Custom Label Content');
			expect(label.text()).not.toBe('Label from prop');
			expect(label.attributes('id')).toBe(
				`${defaultProps.name}-${defaultProps.id}-label`
			);
		});

		it('does not render label when neither prop nor slot is provided', () => {
			wrapper = mount(PdapInputDatePicker, {
				props: {
					name: 'testName',
					id: 'test-id',
					// label prop intentionally omitted
				},
				global: {
					provide: {
						[provideKey as symbol]: {
							setValues: mockSetValues,
							values: mockValues,
							v$: mockV$,
						},
					},
					stubs: {
						VueDatePicker: true,
					},
				},
			});

			expect(wrapper.find('label').exists()).toBe(false);
		});

		it('renders complex label slot content', () => {
			wrapper = mount(PdapInputDatePicker, {
				props: {
					...defaultProps,
				},
				slots: {
					label: `
                  <div class="complex-label">
                      <span class="label-title">Complex Label</span>
                      <small class="label-hint">* Required field</small>
                  </div>
              `,
				},
				global: {
					provide: {
						[provideKey as symbol]: {
							setValues: mockSetValues,
							values: mockValues,
							v$: mockV$,
						},
					},
					stubs: {
						VueDatePicker: true,
					},
				},
			});

			const label = wrapper.find(`label[for="${defaultProps.id}"]`);
			expect(label.exists()).toBe(true);
			expect(label.find('.complex-label').exists()).toBe(true);
			expect(label.find('.label-title').text()).toBe('Complex Label');
			expect(label.find('.label-hint').text()).toBe('* Required field');
		});

		it('renders error slot when slot is provided', () => {
			mockV$.value = {
				...mockV$.value,
				testName: {
					$error: true,
					// @ts-expect-error
					$errors: [{ $message: 'Error Message' }],
				},
			};

			wrapper = mount(PdapInputDatePicker, {
				props: {
					...defaultProps,
				},
				slots: {
					error: '<span class="custom-error">Custom Error Message</span>',
				},
				global: {
					provide: {
						[provideKey as symbol]: {
							setValues: mockSetValues,
							values: mockValues,
							v$: mockV$,
						},
					},
					stubs: {
						VueDatePicker: true,
					},
				},
			});

			const errorElement = wrapper.find(
				'.pdap-input-error-message .custom-error'
			);
			expect(errorElement.exists()).toBe(true);
			expect(errorElement.text()).toBe('Custom Error Message');
		});
	});

	describe('Form Integration', () => {
		it('calls setValues when date changes', async () => {
			const datePicker = wrapper.findComponent(VueDatePicker);
			const newDate = new Date('2024-01-01');

			await datePicker.vm.$emit('update:modelValue', newDate);

			expect(mockSetValues).toHaveBeenCalledWith({
				[defaultProps.name]: newDate,
			});
		});

		it('updates date when form values change externally', async () => {
			const newDate = new Date('2024-01-01');
			mockValues.value = {
				[defaultProps.name]: newDate,
			};

			await wrapper.vm.$nextTick();

			// @ts-expect-error
			expect(wrapper.vm.date).toEqual(newDate);
		});

		it('clears date when form value is removed', async () => {
			mockValues.value = {
				[defaultProps.name]: undefined,
			};

			await wrapper.vm.$nextTick();

			// @ts-expect-error
			expect(wrapper.vm.date).toBeUndefined();
		});
	});

	describe('Dark Mode', () => {
		it('initializes with system dark mode preference', () => {
			const darkModeMatchMedia = vi.fn().mockImplementation(() => ({
				matches: true,
				addEventListener: vi.fn(),
				removeEventListener: vi.fn(),
			}));
			window.matchMedia = darkModeMatchMedia;

			wrapper = mount(PdapInputDatePicker, {
				props: defaultProps,
				global: {
					provide: {
						[provideKey as symbol]: {
							setValues: mockSetValues,
							values: mockValues,
							v$: mockV$,
						},
					},
					stubs: {
						VueDatePicker: true,
					},
				},
			});

			// @ts-expect-error
			expect(wrapper.vm.darkModePreference).toBe(true);
		});

		it('updates dark mode preference when system preference changes', async () => {
			const mockEvent = {
				matches: true,
				type: 'change',
			} as MediaQueryListEvent;

			// @ts-expect-error
			wrapper.vm.updateColorMode(mockEvent);
			await wrapper.vm.$nextTick();

			// @ts-expect-error
			expect(wrapper.vm.darkModePreference).toBe(true);
		});
	});

	describe('Validation', () => {
		it('shows validation error from v$', async () => {
			mockV$.value = {
				testName: {
					$error: true,
					// @ts-expect-error
					$errors: [{ $message: 'Validation error' }],
				},
			};

			await wrapper.vm.$nextTick();

			expect(wrapper.find('.pdap-input-error-message').text()).toBe(
				'Validation error'
			);
		});
	});

	describe('Cleanup', () => {
		it('removes event listener on unmount', async () => {
			wrapper.unmount();

			expect(removeEventListener).toHaveBeenCalledWith(
				'change',
				expect.any(Function)
			);
		});
	});
});
