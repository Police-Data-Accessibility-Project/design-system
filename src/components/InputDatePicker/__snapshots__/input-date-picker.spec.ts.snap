// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PdapInputDatePicker > Rendering > does not render label when neither prop nor slot is provided 1`] = `
<div class="pdap-input">
  <!--v-if-->
  <!--v-if-->
  <vue-date-picker-stub actionrow="[object Object]" arialabels="[object Object]" arrownavigation="false" autoapply="false" autocomplete="off" autoposition="false" canceltext="Cancel" clearable="false" dark="false" disabled="false" disableddates disabledweekdays disablemonthyearselect="false" disableyearselect="false" enableminutes="false" enableseconds="false" enabletimepicker="false" escclose="false" filters="[object Object]" flow focusstartdate="false" hideinputicon="false" hidenavigation hideoffsetdates="false" hoursgridincrement="1" hoursincrement="1" id="test-id" ignoretimevalidation="false" inline="false" is24="false" loading="false" locale="en-Us" markers minutesgridincrement="5" minutesincrement="1" modelauto="false" monthchangeonarrows="false" monthchangeonscroll="false" monthnameformat="short" monthpicker="false" multicalendars="false" multidates="false" name="testName" nohoursoverlay="false" nominutesoverlay="false" nosecondsoverlay="false" notoday="false" nowbuttonlabel="Now" offset="10" partialflow="false" placeholder position="center" presetdates preventminmaxnavigation="false" previewformat quarterpicker="false" range="false" required="false" reverseyears="false" secondsgridincrement="5" secondsincrement="1" selecttext="Select" sixweeks="false" spaceconfirm="false" state="false" teleport="false" teleportcenter="false" textinput="false" timepicker="false" timepickerinline="false" transitions="false" ui="[object Object]" utc="false" vertical="false" weeknumname="W" weekpicker="false" weekstart="1" yearfirst="false" yearpicker="false" yearrange="1900,2100"></vue-date-picker-stub>
</div>
`;

exports[`PdapInputDatePicker > Rendering > renders complex label slot content 1`] = `
<div class="pdap-input">
  <label for="test-id" id="testName-test-id-label">
    <div class="complex-label">
      <span class="label-title">Complex Label</span>
      <small class="label-hint">* Required field</small>
    </div>
  </label>
  <!--v-if-->
  <vue-date-picker-stub actionrow="[object Object]" arialabels="[object Object]" arrownavigation="false" autoapply="false" autocomplete="off" autoposition="false" canceltext="Cancel" clearable="false" dark="false" disabled="false" disableddates disabledweekdays disablemonthyearselect="false" disableyearselect="false" enableminutes="false" enableseconds="false" enabletimepicker="false" escclose="false" filters="[object Object]" flow focusstartdate="false" hideinputicon="false" hidenavigation hideoffsetdates="false" hoursgridincrement="1" hoursincrement="1" id="test-id" ignoretimevalidation="false" inline="false" is24="false" label="Test Label" loading="false" locale="en-Us" markers minutesgridincrement="5" minutesincrement="1" modelauto="false" monthchangeonarrows="false" monthchangeonscroll="false" monthnameformat="short" monthpicker="false" multicalendars="false" multidates="false" name="testName" nohoursoverlay="false" nominutesoverlay="false" nosecondsoverlay="false" notoday="false" nowbuttonlabel="Now" offset="10" partialflow="false" placeholder position="center" presetdates preventminmaxnavigation="false" previewformat quarterpicker="false" range="false" required="false" reverseyears="false" secondsgridincrement="5" secondsincrement="1" selecttext="Select" sixweeks="false" spaceconfirm="false" state="false" teleport="false" teleportcenter="false" textinput="false" timepicker="false" timepickerinline="false" transitions="false" ui="[object Object]" utc="false" vertical="false" weeknumname="W" weekpicker="false" weekstart="1" yearfirst="false" yearpicker="false" yearrange="1900,2100"></vue-date-picker-stub>
</div>
`;

exports[`PdapInputDatePicker > Rendering > renders error slot when slot is provided 1`] = `
<div class="pdap-input pdap-input-error">
  <label for="test-id" id="testName-test-id-label">Test Label</label>
  <div class="pdap-input-error-message">
    <span class="custom-error">Custom Error Message</span>
  </div>
  <vue-date-picker-stub actionrow="[object Object]" arialabels="[object Object]" arrownavigation="false" autoapply="false" autocomplete="off" autoposition="false" canceltext="Cancel" clearable="false" dark="false" disabled="false" disableddates disabledweekdays disablemonthyearselect="false" disableyearselect="false" enableminutes="false" enableseconds="false" enabletimepicker="false" escclose="false" filters="[object Object]" flow focusstartdate="false" hideinputicon="false" hidenavigation hideoffsetdates="false" hoursgridincrement="1" hoursincrement="1" id="test-id" ignoretimevalidation="false" inline="false" is24="false" label="Test Label" loading="false" locale="en-Us" markers minutesgridincrement="5" minutesincrement="1" modelauto="false" monthchangeonarrows="false" monthchangeonscroll="false" monthnameformat="short" monthpicker="false" multicalendars="false" multidates="false" name="testName" nohoursoverlay="false" nominutesoverlay="false" nosecondsoverlay="false" notoday="false" nowbuttonlabel="Now" offset="10" partialflow="false" placeholder position="center" presetdates preventminmaxnavigation="false" previewformat quarterpicker="false" range="false" required="false" reverseyears="false" secondsgridincrement="5" secondsincrement="1" selecttext="Select" sixweeks="false" spaceconfirm="false" state="true" teleport="false" teleportcenter="false" textinput="false" timepicker="false" timepickerinline="false" transitions="false" ui="[object Object]" utc="false" vertical="false" weeknumname="W" weekpicker="false" weekstart="1" yearfirst="false" yearpicker="false" yearrange="1900,2100"></vue-date-picker-stub>
</div>
`;

exports[`PdapInputDatePicker > Rendering > renders label prop when no slot is provided 1`] = `
<div class="pdap-input">
  <label for="test-id" id="testName-test-id-label">Label from prop</label>
  <!--v-if-->
  <vue-date-picker-stub actionrow="[object Object]" arialabels="[object Object]" arrownavigation="false" autoapply="false" autocomplete="off" autoposition="false" canceltext="Cancel" clearable="false" dark="false" disabled="false" disableddates disabledweekdays disablemonthyearselect="false" disableyearselect="false" enableminutes="false" enableseconds="false" enabletimepicker="false" escclose="false" filters="[object Object]" flow focusstartdate="false" hideinputicon="false" hidenavigation hideoffsetdates="false" hoursgridincrement="1" hoursincrement="1" id="test-id" ignoretimevalidation="false" inline="false" is24="false" label="Label from prop" loading="false" locale="en-Us" markers minutesgridincrement="5" minutesincrement="1" modelauto="false" monthchangeonarrows="false" monthchangeonscroll="false" monthnameformat="short" monthpicker="false" multicalendars="false" multidates="false" name="testName" nohoursoverlay="false" nominutesoverlay="false" nosecondsoverlay="false" notoday="false" nowbuttonlabel="Now" offset="10" partialflow="false" placeholder position="center" presetdates preventminmaxnavigation="false" previewformat quarterpicker="false" range="false" required="false" reverseyears="false" secondsgridincrement="5" secondsincrement="1" selecttext="Select" sixweeks="false" spaceconfirm="false" state="false" teleport="false" teleportcenter="false" textinput="false" timepicker="false" timepickerinline="false" transitions="false" ui="[object Object]" utc="false" vertical="false" weeknumname="W" weekpicker="false" weekstart="1" yearfirst="false" yearpicker="false" yearrange="1900,2100"></vue-date-picker-stub>
</div>
`;

exports[`PdapInputDatePicker > Rendering > renders label slot when provided 1`] = `
<div class="pdap-input">
  <label for="test-id" id="testName-test-id-label">
    <span class="custom-label">Custom Label Content</span>
  </label>
  <!--v-if-->
  <vue-date-picker-stub actionrow="[object Object]" arialabels="[object Object]" arrownavigation="false" autoapply="false" autocomplete="off" autoposition="false" canceltext="Cancel" clearable="false" dark="false" disabled="false" disableddates disabledweekdays disablemonthyearselect="false" disableyearselect="false" enableminutes="false" enableseconds="false" enabletimepicker="false" escclose="false" filters="[object Object]" flow focusstartdate="false" hideinputicon="false" hidenavigation hideoffsetdates="false" hoursgridincrement="1" hoursincrement="1" id="test-id" ignoretimevalidation="false" inline="false" is24="false" label="Label from prop" loading="false" locale="en-Us" markers minutesgridincrement="5" minutesincrement="1" modelauto="false" monthchangeonarrows="false" monthchangeonscroll="false" monthnameformat="short" monthpicker="false" multicalendars="false" multidates="false" name="testName" nohoursoverlay="false" nominutesoverlay="false" nosecondsoverlay="false" notoday="false" nowbuttonlabel="Now" offset="10" partialflow="false" placeholder position="center" presetdates preventminmaxnavigation="false" previewformat quarterpicker="false" range="false" required="false" reverseyears="false" secondsgridincrement="5" secondsincrement="1" selecttext="Select" sixweeks="false" spaceconfirm="false" state="false" teleport="false" teleportcenter="false" textinput="false" timepicker="false" timepickerinline="false" transitions="false" ui="[object Object]" utc="false" vertical="false" weeknumname="W" weekpicker="false" weekstart="1" yearfirst="false" yearpicker="false" yearrange="1900,2100"></vue-date-picker-stub>
</div>
`;
